### ISSUE TITLE
 Query by latitude and longitude

### ISSUE THREAD

**Issue by samz406** [ClickHouse/ClickHouse#7775]
Iwant specify a latitude and longitude, given a range value (in km), and find data in the range around latitude and longitude.Is there a calculation formula available?

**Comment by den-crane** [comment]
The same as greatCircleDistance 
https://clickhouse.yandex/docs/en/query_language/functions/geo/#greatcircledistance
Calculate the distance between two points on the Earth's surface.

**Comment by samz406** [comment]
I hava another queustion.source data is str. I use toFloat() convert str to float. left param is convert data,right param is source dat str. they are not equal.Is that correct?
![image](https://user-images.githubusercontent.com/3055810/68926214-f2179080-07bf-11ea-95e5-f40187fb4adb.png)

**Comment by samz406** [comment]
thank's ,the source data is String type. for exmaple ,the lon is '23.739694', I use toFloat convert float.the convert data is not 23.739694.  why is that?

**Comment by rochoa** [comment]
If you need better precision, you could go with toFloat64. See the example below.

```
SELECT 
    toFloat32('23.739694'), 
    toFloat64('23.739694')

┌─toFloat32('23.739694')─┬─toFloat64('23.739694')─┐
│              23.739695 │              23.739694 │
└────────────────────────┴────────────────────────┘

1 rows in set. Elapsed: 0.004 sec. 
```

Beware that computers can not represent _all the numbers_, depending on the precision you use, you might get not-so-perfect results.

```
SELECT toFloat32('0.1') + 1

┌─plus(toFloat32('0.1'), 1)─┐
│        1.1000000014901161 │
└───────────────────────────┘

1 rows in set. Elapsed: 0.008 sec. 
```

A good article in plain English about number representation in computers: https://macwright.org/2017/01/28/numbers.html.
